File: backend/src/main/scala/ch/countrygame/Classes/Country.scala
package ch.countrygame.Classes

import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class Country(id: Int, name: String, details: String, countryCode: String, currencyId: Int, modifierId: Int, manpowerNumber: Option[Double] = None) {
}
object Country {
  implicit val encoder = deriveEncoder[Country]
  implicit val decoder = deriveDecoder[Country]
}

File: frontend/src/app/models/country.ts
export class Country {
  id: number;
  name: string;
  details: string;
  countryCode: string;
  currencyId: number;
  manpowerId: number;
  manpowerNumber?: number;

  constructor(id: number, name: string, details: string, countryCode: string, currencyId: number, manpowerId: number,
              manpowerNumber?: number) {
    this.id = id;
    this.name = name;
    this.details = details;
    this.countryCode = countryCode;
    this.currencyId = currencyId;
    this.manpowerId = manpowerId;
    this.manpowerNumber = manpowerNumber;
  }
}
